### DISCUSSION TITLE
SequentialFeatureSelection using XGBClassifier with montone_constraints

### DISCUSSION THREAD

**Discussion by elchapeau** [rasbt/mlxtend#1053] [Q&A]
I am trying to use SequentialFeatureSelector with XGBClassifer but with added montonic constraints on the features. Below is sample code:

```
params = {'monotone_constraints': '(1,1,1,1,1,-1,-1,-1,-1,-1}'}
model = xgb.XGBClassifier(**params, n_estimators=5)
sfs = SequentialFeatureSelector(model, k_features='best', forward=True, floating=True, scoring='f1_score', cv=5)
sfs = sfs.fit(X_train, y_train)
```

After testing, I think the monotone_constraints are not passed down to various iterations and different columns combinations tried. Is this correct?

I can see why this would be tricky because you would need to make sure that the constraints are maintained in the various iterations as you try different column combinations. Has the functionality to support this been considered? Is there an existing workaround?

Many thanks!

**Comment by rasbt** [comment]
Oh, that's a tricky one. To be honest, I don't have a good answer and workaround for this, sry!
